core/station/impl/src/factories/blockchains/ethereum.rs

```diff
@@ -0,0 +1,295 @@
+                message_hash: tx_signature_hash.clone(),
+                derivation_path: principal_to_derivation_path(&account),
+                key_id: get_key_id(),
+            })
+            .await
+            .expect("failed to sign transaction");
+
+            let sig_bytes = signature.signature.as_slice();
+            let public_key = ecdsa_pubkey_of(&account).await;
+            let parity = y_parity(&tx_signature_hash, sig_bytes, &public_key);
+            alloy::signers::Signature::from_bytes_and_parity(sig_bytes, parity)
+                .expect("failed to decode signature")
+        };
+
+        {
+            // TODO: just to test the recovery. Remove this when done.
+            let test_recovered_address = signature
+                .recover_address_from_prehash(&transaction.signature_hash())
+                .expect("failed to recover address");
+            let account_address = get_address_from_account(&account).await;
+            print(format!(
+                "test_recovered_address: {} {:?} {:?}",
+                hex::encode_prefixed(test_recovered_address).to_lowercase()
+                    == account_address.to_lowercase(),
+                hex::encode_prefixed(test_recovered_address),
+                account_address
+            ));
+        }
+
+        print(format!("signature: {:?}", signature));
+        let tx_signed = transaction.into_signed(signature);
+        let tx_envelope: alloy::consensus::TxEnvelope = tx_signed.into();
+        let tx_encoded = tx_envelope.encoded_2718();
+        print(format!("tx_encoded: {:?}", tx_encoded));
+
+        let sent_tx_hash = eth_send_raw_transaction(&self.chain, &tx_encoded).await;
+        print(format!("sent_tx_hash: {:?}", sent_tx_hash));
+
+        Ok(BlockchainTransactionSubmitted {
+            details: vec![(
+                TRANSACTION_SUBMITTED_DETAILS_TRANSACTION_HASH_KEY.to_owned(),
+                sent_tx_hash,
+            )],
+        })
+    }
+}
+
+async fn ecdsa_pubkey_of(account: &Account) -> Vec<u8> {
+    let (key,) = ecdsa_public_key(EcdsaPublicKeyArgument {
+        canister_id: None,
+        derivation_path: principal_to_derivation_path(&account),
+        key_id: get_key_id(),
+    })
+    .await
+    .expect("failed to get public key");
+    key.public_key
+}
+
+async fn get_address_from_account(account: &Account) -> String {
+    let public_key = ecdsa_pubkey_of(&account).await;
+    let address = get_address_from_public_key(&public_key);
+    hex::encode_prefixed(&address)
+}
+
+fn get_address_from_public_key(public_key: &[u8]) -> Address {
+    let verifying_key = ecdsa::VerifyingKey::from_sec1_bytes(&public_key)
+        .expect("Failed to create VerifyingKey from public key bytes");
+    alloy::signers::utils::public_key_to_address(&verifying_key)
+}
+
+fn get_key_id() -> EcdsaKeyId {
+    // TODO: check what we should use as a name
+    let name: String = "dfx_test_key".to_string();
```

Comment:
We should provide the key id in a different manner since we different different keys based on mainnet vs local dfx runtime